src/plugins/intel_cpu/src/emitters/jit_snippets_emitters.cpp

```diff
@@ -126,263 +157,211 @@ void KernelEmitter::validate_arguments(const std::vector<size_t> &in,
-    } else if (outer_work_amount > 1) {
-        // We need to create a Loop in this case
-        h->mov(reg_outer_amount, outer_work_amount);
-        h->L(for_body);
-        {
-            emit_tiles(reg_inner_amount, data_ptr_regs, vector_size, vec_pool, local_gpr_pool);
-
-            // Todo: Load and Store emitters are currently implemented so they ALWAYS increment appropriate pointers
-            //   after reading/writing. This might be a problem if we need to read the same data multiple times (broadcasting shapes).
-            //   To overcome this limitation, we add appropriate negative offsets if necessary.
-            for (auto i = 0; i < num_params; i++) {
-                if (jcp.scheduler_offsets[i] != 0) {
-                    h->add(data_ptr_regs[i], jcp.scheduler_offsets[i]);
-                }
-            }
-            // Note that outer dimensions are always incremented by 1 (outer tiles are always scalar)
-            h->sub(reg_outer_amount, 1);
-            h->cmp(reg_outer_amount, 1);
-            h->jge(for_body, CodeGenerator::T_NEAR);
-        }
+    // save previous register state (if there is an outer loop that uses this reg for example)
+    if (!evaluate_once) {
+        h->mov(reg_work_amount, work_amount);
     }
+    // Note: loop address is not calculated at this point, so need to call calcJmpAddress() which is protected
+    // or ready(), but they both set internal flags and that's not a desired way to use them.
+    // So the most obvious WA is just to use current address manually
+    loop_begin->begin_address = h->getCurr();
+    loop_begin->input_regs = in;
+}
+
+LoopEndEmitter::LoopEndEmitter(dnnl::impl::cpu::x64::jit_generator* h, dnnl::impl::cpu::x64::cpu_isa_t isa,
+                                   const std::shared_ptr<ov::Node>& n) : jit_emitter(h, isa, n) {
+    loop_end = ov::as_type_ptr<ngraph::snippets::op::LoopEnd>(n);
+    if (!loop_end)
+        IE_THROW() << "LoopEndEmitter invoked with invalid op argument";
+    loop_begin = loop_end->get_loop_begin();
+    // todo: this check could be excessive, since we check for it in validate_and_infer_types()
+    if (!loop_begin)
+        IE_THROW() << "LoopEndEmitter invoked with invalid configuration: the last arg must be LoopBegin";
+    // Note that 1 edge connects LoopBegin and LoopEnd
+    num_inputs = loop_begin->get_input_size();
+    num_outputs = loop_end->get_output_size();
+    increment = loop_end->get_increment();
+    work_amount = loop_end->get_work_amount();
+    apply_increments = loop_end->get_apply_increment();
+    finalization_offsets = loop_end->get_finalization_offsets();
+    evaluate_once = loop_end->get_evaluate_once();
+    for (int i = 0; i < num_inputs; i++)
+        io_data_size.push_back(loop_begin->get_input_element_type(i).size());
+    for (int i = 0; i < num_outputs; i++)
+        io_data_size.push_back(loop_end->get_input_element_type(i).size());
```

Comment:
get_output_element_type?

Reply:
It doesn't really matters from functional perspective, since LoopEnd bypasses tesnsors from input to outputs.
But I agree that this would make our intentions more clear and thus increase code readability.
corrected.